rotkehlchen/globaldb/cache.py

```diff
@@ -122,3 +122,18 @@ def globaldb_delete_general_cache(
         write_cursor.execute('DELETE FROM general_cache WHERE key=?', (cache_key,))
     else:
         write_cursor.execute('DELETE FROM general_cache WHERE key=? AND value=?', (cache_key, value))  # noqa: E501
+
+
+def globaldb_delete_general_cache_like(
+        write_cursor: DBCursor,
+        key_parts: Iterable[Union[str, GeneralCacheType]],
+) -> None:
+    """Function to delete globaldb cache. Deletes all the values that start with key."""
+    condition = all(
+        isinstance(part, GeneralCacheType) or ('%' not in part and '.' not in part)
+        for part in key_parts
+    )
+    message = 'Parts of cache key that is used with LIKE should not contain "%" or "."'
+    assert condition, message
```

Comment:
The  issue with moving this to a variable is that that the check doesn't get removed when building for production and you are adding extra logic. It should be inside the assert otherwise you are making the check always